aztec/src/main/kotlin/org/wordpress/aztec/formatting/IndentFormatter.kt

```diff
@@ -0,0 +1,186 @@
+package org.wordpress.aztec.formatting
+
+import org.wordpress.aztec.AztecText
+import org.wordpress.aztec.spans.AztecDynamicImageSpan
+import org.wordpress.aztec.spans.AztecHeadingSpan
+import org.wordpress.aztec.spans.AztecPreformatSpan
+import org.wordpress.aztec.spans.AztecQuoteSpan
+import org.wordpress.aztec.spans.IAztecBlockSpan
+import org.wordpress.aztec.spans.ParagraphSpan
+
+/**
+ * Class that handles indents of text blocks
+ */
+class IndentFormatter(editor: AztecText) : AztecFormatter(editor) {
+    /**
+     * This function tries to indent all the selected lines that can be indented. It means adding a `\t` character
+     * at the beginning of the line (after the `\n` which marks the end of previous line).
+     */
+    fun indent() {
+        // Contains all the indices which should be indented at once to avoid unnecessary redrawing
+        val indicesToIndent = mutableSetOf<Int>()
+        // Look for the line break before the selection start
+        val previousLineBreak = editableText.substring(0, selectionStart).lastIndexOf("\n") + 1
+        var firstLineIndented = false
+        // Check whether the current line can be indented and indents it if necessary
+        if (selectionCanBeIndented(previousLineBreak, selectionStart)) {
+            firstLineIndented = true
+            indicesToIndent.add(previousLineBreak)
+        }
```

Comment:
This can be simplified with:

```suggestion
        val firstLineIndented = if (selectionCanBeIndented(previousLineBreak, selectionStart)) {
            indicesToIndent.add(previousLineBreak)
            true
        } else false
```